### PR TITLE
AtomsBase support

### PR DESCRIPTION
**Pull request by rkurchin**
Adds support for building graphs from AtomsBase systems. See commit message for more details...

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -17,6 +18,8 @@ NearestNeighbors = "b8a86587-4115-5ab1-83bc-aa920d37bbce"
 PyCall = "438e738f-606a-5dbb-bf0a-cddfbfd45ab0"
 Serialization = "9e88b42a-f829-5b0c-bbe9-9e923198166b"
 SimpleWeightedGraphs = "47aef6b3-ad0c-573a-a1e2-d07658019622"
+StaticArrays = "90137ffa-7385-5640-81b9-e52037218182"
+Unitful = "1986cc42-f94f-5a68-af5c-568840ba703d"
 Xtals = "ede5f01d-793e-4c47-9885-c447d1f18d6d"
 
 [compat]
```
Comment (thazhemadam) [inline Project.toml:25]: We should add compats for `AtomsBase`, `StaticArrays` and `Unitful`

File: src/graph_building.jl [hunk 2]
```diff
@@ -49,35 +52,37 @@ function build_graph(
...
+- `sys`: either an Xtals Crystal object or an AtomsBase AbstractSystem representing the atomic geometry from which to build a graph
 
 ## Keyword Arguments
 - `cutoff_radius::Real=8.0`: cutoff radius for atoms to be considered neighbors (in angstroms)
 - `max_num_nbr::Integer=12`: maximum number of neighbors to include (even if more fall within cutoff radius)
 - `dist_decay_func::Function=inverse_square`: function to determine falloff of graph edge weights with neighbor distance
 """
 function build_graph(
-    crys::Crystal;
+    sys;
```
Comment (thazhemadam) [inline src/graph_building.jl:71]:
```suggestion
    sys::Union{Crystal, AtomsBase.AbstractSystem};
```

File: src/graph_building.jl [hunk 3]
```diff
@@ -145,33 +150,44 @@ function weights_voronoi(struc)
...
-Find all lists of pairs of atoms in `crys` that are within a distance of `cutoff_radius` of each other, respecting periodic boundary conditions.
+Find all lists of pairs of atoms in sys (which can be an Xtals.Crystal or any AtomsBase AbstractSystem object) that are within a distance of `cutoff_radius` of each other, respecting periodic boundary conditions.
 
 Returns as is, js, dists to be compatible with ASE's output format for the analogous function.
 """
-function neighbor_list(crys::Crystal; cutoff_radius::Real = 8.0)
-    n_atoms = crys.atoms.n
-
-    # make 3 x 3 x 3 supercell and find indices of "middle" atoms
+function neighbor_list(sys; cutoff_radius::Real = 8.0)
```
Comment (thazhemadam) [inline src/graph_building.jl:174]:
```suggestion
function neighbor_list(sys::Union{Crystal, AtomsBase.AbstractSystem}; cutoff_radius::Real = 8.0)
```

**Comment by rkurchin** [comment]
@thazhemadam, didn't we see this GLIBCXX thing before with another Python dependency somewhere? I can't recall what the fix was...

**Comment by rkurchin** [comment]
I was wondering if switching over to PythonCall might help any of these issues (I've heard it's better than PyCall, also that it doesn't play well with PyCall which is presumably related to why I couldn't get AtomsIO.jl to play nice with this, which I was also hoping to use for reading/writing files), but am having trouble getting that working either. If you want to take a peek, check out the `pythoncall` branch I just pushed...

**Review by thazhemadam — CHANGES_REQUESTED** [review]
Mostly looks good to me. Just a few suggestions. 🙂

**Comment by thazhemadam** [comment]
> @thazhemadam, didn't we see this GLIBCXX thing before with another Python dependency somewhere? I can't recall what the fix was...

Yeah, I vaguely recall seeing this too. I think using `build/deps.jl` helped fix the issue? Not entirely sure.

**Comment by thazhemadam** [comment]
>  I was wondering if switching over to PythonCall might help any of these issues (I've heard it's better than PyCall, also that it doesn't play well with PyCall which is presumably related to why I couldn't get AtomsIO.jl to play nice with this, which I was also hoping to use for reading/writing files), but am having trouble getting that working either. If you want to take a peek, check out the pythoncall branch I just pushed...


PythonCall _might_ help with these issues, but it will also start making trouble any time you try to use sysimages with the packages, which is why I'm wary of it. [^1] I also have a branch which has `PythonCall` + `CondaPkg` set up - [`at/condapkg`](https://github.com/Chemellia/AtomGraphs.jl/tree/at/condapkg). 😆
I didn't open a PR since I imagine being unable to package this is a bit of a deal breaker.

[^1]: https://github.com/cjdoris/PythonCall.jl/issues/129

**Comment by rkurchin** [comment]
Cherry-picking this over to #10, closing...

**Pull request closed without merging**